packages/cli/src/commands/__tests__/commands.spec.ts

```diff
@@ -0,0 +1,113 @@
+import * as prismHttp from '@stoplight/prism-http';
+import * as yargs from 'yargs';
+import { createMultiProcessPrism, createSingleProcessPrism } from '../../util/createServer';
+import mockCommand from '../mock';
+import proxyCommand from '../proxy';
+
+const parser = yargs.command(mockCommand).command(proxyCommand);
+
+jest.mock('../../util/createServer', () => ({
+  createMultiProcessPrism: jest.fn().mockResolvedValue([]),
+  createSingleProcessPrism: jest.fn().mockResolvedValue([]),
+}));
+
+jest.spyOn(prismHttp, 'getHttpOperationsFromResource').mockResolvedValue([]);
+
+describe.each<{ 0: string; 1: string; 2: unknown }>([
+  ['mock', '', { dynamic: false }],
+  ['proxy', 'http://github.com', { upstream: new URL('http://github.com/') }],
+])('%s command', (command, upstream, additionalCallOptions) => {
+  beforeEach(() => {
+    (createSingleProcessPrism as jest.Mock).mockClear();
+    (createMultiProcessPrism as jest.Mock).mockClear();
+  });
+
+  test(`starts ${command} server`, () => {
+    parser.parse(`${command} /path/to ${upstream}`);
+
+    expect(createMultiProcessPrism).not.toHaveBeenCalled();
+    expect(createSingleProcessPrism).toHaveBeenLastCalledWith({
+      document: '/path/to',
+      multiprocess: false,
+      reportViolations: 'stdout',
+      cors: true,
+      host: '127.0.0.1',
+      port: 4010,
+      ...additionalCallOptions,
+    });
+  });
+
+  test(`starts ${command} server on custom port`, () => {
+    parser.parse(`${command} /path/to -p 666 ${upstream}`);
+
+    expect(createMultiProcessPrism).not.toHaveBeenCalled();
+    expect(createSingleProcessPrism).toHaveBeenLastCalledWith({
```

Comment:
I'm thinking of changing those expectations into something more like:
`expect(createSingleProcessPrism).toHaveBeenLastCalledWith(expect.objectContaining({ port: 666 }))`. Less code, almost the same confidence.

Reply:
Very good point. Addressed in 88da6af